### DISCUSSION TITLE
Allow for adding address during user registration

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#12389] [Ideas] [answered]
As far I understand, right now user needs to be registered to address. 
It would be useful If we could already add the first address using the same registration form.

**Comment by piotrpog** [comment]
Thought i would bump this - recently had a craft commerce client that needed this functionality.
Most webshops allow setting address when registering, so this could be useful for craft too.

**Comment by brandonkelly** [comment, marked as answer]
The `users/save-user` action is already pretty involved so I’m not sure I want to add additional complexity to it, unrelated to actually saving the user.

But you could get this working yourself by adding a custom controller/action to a module that calls `users/save-user` internally, and then checks for the address data and saves that as well.

```php
use craft\elements\Address;
use craft\elements\User;
use yii\web\Response;

public function actionCreateUser(): ?Response
{
    /** @var Response|null $response */
    $response = $this->run('users/save-user');

    if ($response?->getStatusCode() !== 200) {
        return $response;
    }

    $addressData = $this->request->getBodyParam('address');
    if (!$addressData) {
        return $response;
    }

    $email = trim($this->request->getBodyParam('email') ?? '');
    if (!$email) {
        return $response;
    }

    $user = User::find()->email($email)->status(null)->one();
    if (!$user) {
        return $response;
    }

    $address = new Address([
        'ownerId' => $user->id,
    ]);
    unset($addressData['id'], $addressData['uid'], $addressData['ownerId'], $addressData['primaryOwnerId']);
    $address->setAttributes($addressData);
    $address->setFieldValuesFromRequest('address.fields');
    Craft::$app->getElements()->saveElement($address);

    return $response;
}
```
